cEP-0019.md

```diff
@@ -0,0 +1,454 @@
+| Status   | Proposed                                   |
+| Type     | Process                                    |
+
+## Abstract
+
+This cEP describes the details of the process of implementing the meta-review
+system as a part of the
+[GSoC'18 project](https://summerofcode.withgoogle.com/projects/#5188493739819008).
+
+## Background
+
+People respond to review comments on pull requests by giving feedback. Emojis
+make it much easier to give and receive feedback without a long comment thread.
+Those emojis are called
+[reactions](https://developer.github.com/v4/enum/reactioncontent/) on GitHub
+or [award emoji](https://docs.gitlab.com/ee/user/award_emojis.html) on GitLab.
+They are used by people to publicly express their feelings to review comments,
+which provides feedback to reviewers and all other members.
+
+GitLab is beyond the scope of this project, as GitHub and GitLab have quite
+different APIs and most core projects of coala are based on GitHub. This
+proposal will only discuss reactions on GitHub, but not award emoji on GitLab.
+
+There are 6 types of reactions on GitHub:
+
+| Name        | Markdown    | Unicode | Emoji |
+| ----------- | ----------- | ------- | ----- |
+| THUMBS_UP   | `:+1:`      | U+1F44D | 👍    |
+| THUMBS_DOWN | `:-1:`      | U+1F44E | 👎    |
+| LAUGH       | `:smile:`   | U+1F604 | 😄    |
+| HOORAY      | `:tada:`    | U+1F389 | 🎉    |
+| CONFUSED    | `:confused` | U+1F615 | 😕    |
+| HEART       | `:heart:`   | U+FE0F  | ❤️    |
+
+There are three types of reviews on a pull request:
+
+1. Issue Comment
+   [see sample](https://github.com/coala/gh-board/pull/18#issuecomment-385878735)
+
+2. Review Summary
+   [see sample](https://github.com/coala/gh-board/pull/18#pullrequestreview-116783753)
+
+3. Review Discussion
+   [see sample](https://github.com/coala/gh-board/pull/18#discussion_r186249423)
+
+However, those reactions are used in an adhoc way. They spread over all PRs
+and are not collected and analyzed. Moreover, GitHub doesn't send
+notifications to reviewers who receive reactions. By tracking reactions, a nice
+feedback loop would be created for both senior and junior reviewers.
+
+## Introduction
+
+The meta-review project is about making use of reactions to build a
+meta-review system, similar to a meta-moderation system. These responses
+are to be collected, processed and displayed on
+[gh-board](https://github.com/coala/gh-board), which is a nice kanban
+board. Compared to [Huboard](https://huboard.com/) which requires server-side
+support, gh-board is serverless.
+
+The scope of this project would exclude non-public meta-reviews. That is,
+there would not be any anonymous meta-reviews as that introduces too many
+additional complexities. The public only version would be used for
+a long time before adding support for non-public meta-reviews.
+
+Note there is a technical limitation in that reactions can not be placed
+onto the "review summary", which means general review comments should not
```

Comment:
'should not'

how is that going to be enforced?

Reply:
It's just a suggestion which will not be enforced. I'll change the wording here.